test_conformance/api/test_queries.cpp

```
@@ -656,6 +656,57 @@ REGISTER_TEST(get_device_info)
     return 0;
 }
 
+REGISTER_TEST(get_device_info_comparability)
+{
+    int error = CL_SUCCESS;
+
+    // comparability test for CL_DEVICE_PLATFORM
+    // 1. find platform related to device without using query
+    cl_uint num_platforms = 0;
+    error = clGetPlatformIDs(16, nullptr, &num_platforms);
+    test_error(error, "clGetPlatformIDs failed");
+
+    std::vector<cl_platform_id> platforms(num_platforms);
+
+    error = clGetPlatformIDs(num_platforms, platforms.data(), &num_platforms);
+    test_error(error, "clGetPlatformIDs failed");
+
+    cl_uint num_devices = 0;
+    cl_platform_id comp_platform = nullptr;
+    for (int p = 0; p < (int)num_platforms && comp_platform == nullptr; p++)
+    {
+        error = clGetDeviceIDs(platforms[p], CL_DEVICE_TYPE_ALL, 0, nullptr,
+                               &num_devices);
+        test_error(error, "clGetDeviceIDs failed");
+
+        std::vector<cl_device_id> devices(num_devices);
+        error = clGetDeviceIDs(platforms[p], CL_DEVICE_TYPE_ALL, num_devices,
+                               devices.data(), nullptr);
+        test_error(error, "clGetDeviceIDs failed");
```

I think we should ignore platforms where these queries return an error vs. failing the test.  I know this is atypical and not the pattern used by most tests, but I think it is the right thing to do here for two reasons:

1. If we fail the test when these queries return an error, it means that a buggy platform that just so happens to be installed on the system alongside the platform being tested could cause the test to fail, even if the platform being tested is correct and conformant.
2. `clGetDeviceIDs` is a little weird in that it is specified to return an error when no devices are found rather than returning `num_device` equal to zero.  This means that a conformant platform could return an error, which would also cause the test to fail.  Linking in: https://github.com/KhronosGroup/OpenCL-Docs/issues/1269

Since this test is run through the harness, we could also consider using the `CL_PLATFORM_INDEX` environment variable to choose the platform to test against, though this loop is fine too, and it has the added benefit of testing that `clGetDeviceIDs` returns the same set of device IDs if it is called multiple times.

---

Done, moreover I've added fallback path based on `CL_PLATFORM_INDEX`, thanks for the hint.